### ISSUE TITLE
jointly training with other models?

### ISSUE THREAD

**Issue by chris-tkinter** [manujosephv/pytorch_tabular#53]
**Is your feature request related to a problem? Please describe.**
Hi, awesome project!

I am thinking about training tabular model jointly with other models. For example, something like 
```
x1 = tabtransformer(data)
x2 = othermodel(data)
x = x1 + x2
loss = criterion(x, label)
```
so that the gradient would flow through both the `othermodel` and `tabtransformer`. I am wondering if this is possible. Currently I can use `DeepFeatureExtractor` to generate the embedding, but this wouldn't update `tabtransformer`'s weights.
**Describe the solution you'd like**
Being able to train tabular model jointly other than `.fit` method.

Thanks!

**Comment by manujosephv** [comment]
If you are open to a bit of coding, you can easily do this. There is a `BaseModel`, which you can subclass and implement `def _build_network(self)` , `def forward(self, x: Dict)`, and override any other method you need to.. like `calculate loss` ,etc. I guess you can check the tutorial [here](https://pytorch-tabular.readthedocs.io/en/latest/tutorials/04-Implementing%20New%20Architectures/).

Once you have a new model, then you can use it just like any other model in PyTorch Tabular

**Comment by chris-tkinter** [comment]
Oh I see, thanks!!
